staging/src/k8s.io/legacy-cloud-providers/openstack/metadata.go

```diff
@@ -67,6 +84,13 @@ type DeviceMetadata struct {
 	// .. and other fields.
 }
 
+type ec2Metadata struct {
+	PublicIP     string `json:"public-ipv4"`
+	LocalIP      string `json:"local-ipv4"`
```

Comment:
We need to also fetch IPv6 addresses from somewhere.  As written, I think this PR will remove support for v6 node addresses in dual-stack/v6-only clusters.

https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/using-instance-addressing.html#view-ipv6-addresses says that EC2 exposes Ipv6 addresses in http://169.254.169.254/latest/meta-data/network/interfaces/macs/:mac_address/ipv6s - does the openstack compatibility version also expose the v6 addresses there/somewhere?